Build failure: kbdd

### Nixpkgs version

- Unstable (26.05)

### Steps to reproduce

`nix build github:NixOS/nixpkgs/1412caf7bf9e660f2f962917c14b1ea1c3bc695e#kbdd`

### Can Hydra reproduce this build failure?

Yes, Hydra can reproduce this build failure.

### Link to Hydra build job

https://hydra.nixos.org/job/nixos/unstable/nixpkgs.kbdd.x86_64-linux

### Relevant log output

```console
gcc -DHAVE_CONFIG_H -I. -I..    -I/nix/store/j359ndglk11v5j47ra66lc1vxlk2dbjw-glib-2.86.3-dev/include/glib-2.0 -I/nix/store/f7rcazhd826xlcz43il4vafv28888cgj-glib-2.86.3/lib/glib-2.0/include -I/nix/store/ylshl2ahw>
In file included from /nix/store/f7rcazhd826xlcz43il4vafv28888cgj-glib-2.86.3/lib/glib-2.0/include/glibconfig.h:9,
                 from /nix/store/j359ndglk11v5j47ra66lc1vxlk2dbjw-glib-2.86.3-dev/include/glib-2.0/glib/gtypes.h:34,
                 from /nix/store/j359ndglk11v5j47ra66lc1vxlk2dbjw-glib-2.86.3-dev/include/glib-2.0/glib/galloca.h:34,
                 from /nix/store/j359ndglk11v5j47ra66lc1vxlk2dbjw-glib-2.86.3-dev/include/glib-2.0/glib.h:32,
                 from kbdd.c:34:
kbdd.c: In function 'main':
/nix/store/j359ndglk11v5j47ra66lc1vxlk2dbjw-glib-2.86.3-dev/include/glib-2.0/glib/gmacros.h:1301:33: warning: Deprecated pre-processor symbol
 1301 | #define _GLIB_GNUC_DO_PRAGMA(x) _Pragma(G_STRINGIFY (x))
      |                                 ^~~~~~~
/nix/store/j359ndglk11v5j47ra66lc1vxlk2dbjw-glib-2.86.3-dev/include/glib-2.0/glib/gmacros.h:1302:31: note: in expansion of macro '_GLIB_GNUC_DO_PRAGMA'
 1302 | #define GLIB_DEPRECATED_MACRO _GLIB_GNUC_DO_PRAGMA(GCC warning "Deprecated pre-processor symbol")
      |                               ^~~~~~~~~~~~~~~~~~~~
/nix/store/j359ndglk11v5j47ra66lc1vxlk2dbjw-glib-2.86.3-dev/include/glib-2.0/glib/glib-visibility.h:141:39: note: in expansion of macro 'GLIB_DEPRECATED_MACRO'
  141 | #define GLIB_DEPRECATED_MACRO_IN_2_32 GLIB_DEPRECATED_MACRO
      |                                       ^~~~~~~~~~~~~~~~~~~~~
/nix/store/j359ndglk11v5j47ra66lc1vxlk2dbjw-glib-2.86.3-dev/include/glib-2.0/glib/deprecated/gthread.h:280:38: note: in expansion of macro 'GLIB_DEPRECATED_MACRO_IN_2_32'
  280 | #define g_thread_supported()     (1) GLIB_DEPRECATED_MACRO_IN_2_32
      |                                      ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~
kbdd.c:305:11: note: in expansion of macro 'g_thread_supported'
  305 |     if ( !g_thread_supported () ) {
      |           ^~~~~~~~~~~~~~~~~~
kbdd.c:187:40: warning: '%s' directive argument is null [-Wformat-overflow=]
  187 |     printf(":main RequestName returned %s.\n", request_ret);
      |                                        ^~
gcc -DHAVE_CONFIG_H -I. -I..    -I/nix/store/j359ndglk11v5j47ra66lc1vxlk2dbjw-glib-2.86.3-dev/include/glib-2.0 -I/nix/store/f7rcazhd826xlcz43il4vafv28888cgj-glib-2.86.3/lib/glib-2.0/include -I/nix/store/ylshl2ahw>
gcc -DHAVE_CONFIG_H -I. -I..    -I/nix/store/j359ndglk11v5j47ra66lc1vxlk2dbjw-glib-2.86.3-dev/include/glib-2.0 -I/nix/store/f7rcazhd826xlcz43il4vafv28888cgj-glib-2.86.3/lib/glib-2.0/include -I/nix/store/ylshl2ahw>
libkbdd.c:223:1: error: conflicting types for 'kbdd_default_loop'; have 'void *(Display *)'
  223 | kbdd_default_loop(Display * display)
      | ^~~~~~~~~~~~~~~~~
In file included from libkbdd.c:29:
libkbdd.h:60:8: note: previous declaration of 'kbdd_default_loop' with type 'void *(void)'
   60 | void * kbdd_default_loop();
      |        ^~~~~~~~~~~~~~~~~
```

### Additional context

It seems that this is already fixed in upstream, namely here: https://github.com/qnikst/kbdd/commit/b87e44afd5859157245eee22b11827605bfa09b9

So it is sufficient to update to the current master revision

### System metadata

 - system: `"x86_64-linux"`
 - host os: `Linux 6.12.63, NixOS, 26.05 (Yarara), 26.05.20251225.3e2499d`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.31.2+1`
 - nixpkgs: `/nix/store/0lc6kbmhkqk8vmgflpn2n1d1qb3vl2f5-source`

### Notify maintainers



---

**Note for maintainers:** Please tag this issue in your pull request description. (i.e. `Resolves #ISSUE`.)


### I assert that this issue is relevant for Nixpkgs

- [x] I assert that this is a bug and not a support request.
- [x] I assert that this is not a [duplicate of an existing issue](https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+label%3A%220.kind%3A+build+failure%22).
- [x] I assert that I have read the [NixOS Code of Conduct](https://github.com/NixOS/.github/blob/master/CODE_OF_CONDUCT.md) and agree to abide by it.

### Is this issue important to you?

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

--- a/pkgs/by-name/kb/kbdd/package.nix
+++ b/pkgs/by-name/kb/kbdd/package.nix
@@ -10,13 +10,13 @@
 
 stdenv.mkDerivation {
   pname = "kbdd";
-  version = "unstable-2021-04-26";
+  version = "unstable-2025-08-10";
 
   src = fetchFromGitHub {
     owner = "qnikst";
     repo = "kbdd";
-    rev = "3145099e1fbbe65b27678be72465aaa5b5872874";
-    sha256 = "1gzcjnflgdqnjgphiqpzwbcx60hm0h2cprncm7i8xca3ln5q6ba1";
+    rev = "b87e44afd5859157245eee22b11827605bfa09b9";
+    hash = "sha256-cbMcB6jgssfMUjemBOiE06zJK2TbzOWt1Rvt41V33Mo=";
   };
 
   nativeBuildInputs = [

